xbmc/network/upnp/UPnPInternal.cpp

```diff
@@ -590,11 +590,13 @@ BuildObject(CFileItem&                    item,
         // finally apply the found artwork
         thumb = item.GetArt("thumb");
         if (!thumb.empty()) {
+            std::string wrappedUrl = CTextureUtils::GetWrappedImageURL(thumb);
+            std::string file_path = wrappedUrl[wrappedUrl.size() - 1] == '/' ? wrappedUrl.erase(wrappedUrl.size() - 1) : wrappedUrl;
```

Comment:
I'm pretty sure we have a utility method for this in `URIUtils` or something like that.

Reply:
Yes there is, URIUtils::RemoveSlashAtEnd :-)
Moved the code to BuildSafeResourceUri, conditional to an image protocol.